ui/apps/platform/src/Containers/PolicyCategories/CreatePolicyCategoryModal.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import * as yup from 'yup';
+import {
+    Modal,
+    ModalVariant,
+    ModalBoxBody,
+    ModalBoxFooter,
+    Button,
+    Form,
+    FormGroup,
+    TextInput,
+} from '@patternfly/react-core';
+import { FormikProvider, useFormik } from 'formik';
+
+import { PolicyCategory } from 'types/policy.proto';
+import { postPolicyCategory } from 'services/PolicyCategoriesService';
+
+type CreatePolicyCategoryModalType = {
+    isOpen: boolean;
+    onClose: () => void;
+    addToast: (toast) => void;
+    refreshPolicyCategories: () => void;
+    setSelectedCategory: (category: PolicyCategory) => void;
+};
+
+const emptyPolicyCategory = {
+    id: '',
+    name: '',
+    isDefault: false,
+};
+
+function CreatePolicyCategoryModal({
+    isOpen,
+    onClose,
+    addToast,
+    refreshPolicyCategories,
+    setSelectedCategory,
+}: CreatePolicyCategoryModalType) {
+    const formik = useFormik({
+        initialValues: emptyPolicyCategory as PolicyCategory,
+        onSubmit: (values, { setSubmitting, resetForm }) => {
+            setSubmitting(false);
+            postPolicyCategory(values)
+                .then(() => {
+                    setTimeout(refreshPolicyCategories, 200);
+                    addToast('Successfully added category');
+                })
+                .catch((error) => {
+                    addToast(error.message);
+                })
+                .finally(() => {
+                    setSubmitting(false);
+                    setSelectedCategory(values);
+                    resetForm();
+                    onClose();
+                });
+        },
+        validateOnMount: true,
+        validationSchema: yup.object().shape({
+            name: yup
+                .string()
+                .min(5, 'Policy category name must be at least 5 characters long')
+                .max(128, 'Policy category name must be less than 128 characters long')
```

Comment:
Is this max-length a backend requirement?
(128 seems like a random choice. I know relationship SQL, like Postgres, used to be optimized for a short string of up to 255 characters--haven't seen 128 before.)

Reply:
Yes, backend requirement:

https://github.com/stackrox/stackrox/blob/master/central/policycategory/service/service_impl.go#L40